doc/subsystems/networking/vlan.rst

```diff
@@ -0,0 +1,49 @@
+.. _vlan-support:
+
+Virtual LAN (VLAN) Support
+##########################
+
+Overview
+********
+
+`Virtual LAN <https://wikipedia.org/wiki/Virtual_LAN/>`_ (VLAN) is a
+partitioned and isolated computer network at the data link layer
+(OSI layer 2). For ethernet network this refers to
+`IEEE 802.1Q <https://en.wikipedia.org/wiki/IEEE_802.1Q/>`_
+
+In Zephyr, each individual VLAN is modelled as a virtual network interface.
+This means that there is a ethernet network interface that corresponds to
+real physical ethernet port in the system. A virtual network interface is
+created for each VLAN, and this virtual network interface connects to the
+real network interface. This is similar as how Linux implements VLANs. The
+*eth0* is the real network interface and *vlan0* is a virtual network interface
+which is run on top of *eth0*.
+
+The VLAN support must be enabled at compile time. The option to set is
+:option:`CONFIG_NET_VLAN`. It is also important that user sets
+:option:`CONFIG_NET_VLAN_COUNT` option correctly. It should be set by the
+user and its value should reflect how many network interfaces there will
+be in the system. For example, the :option:`CONFIG_NET_VLAN_COUNT` should
+be set to 3 (1 + 2) if there is one network interface without VLAN support,
+and two network interfaces with VLAN support. In practice the value of
+:option:`CONFIG_NET_VLAN_COUNT` should be set to same as total number of
+network interfaces in the system.
```

Comment:
That part is a bit confusing but I guess it's due to how net_if (real vs virtual) instances are generated in case of VLAN support.

Reply:
Yes, if VLAN support is disabled, then system will automatically handle the amount of network interfaces at runtime (no compile time configuration is needed).
If VLAN support is enabled, then we need some information about number of interfaces at compile time, and we do not know this information at this point thus the need for this info as a config var.